ui/src/geometry/geometry-api.ts

```diff
@@ -63,7 +65,7 @@ const planVerticalGeometryCache = asyncCache<
     PlanVerticalGeometryKey,
     VerticalGeometryItem[] | undefined
 >;
-type LocationTrackVerticalGeometryKey = `${LocationTrackId}_${PublishType}`;
+type LocationTrackVerticalGeometryKey = `${LocationTrackId}_${PublicationState}_${LayoutDesignId}`;
```

Comment:
Myös cache-avaimissa on pyritty ottamaan huomioon koko `LayoutContext` ettei myöhemmin tule ongelmia kakkuinvalidaation kanssa. Jos näin ei ole, niin saa älähtää